### ISSUE TITLE
Monarc install from source fails

### ISSUE THREAD

**Issue by reuz** [monarc-project/MonarcAppFO#257]
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 79 installs, 0 updates, 0 removals
  - Installing laminas/laminas-dependency-plugin (1.0.3): Loading from cache
PHP Fatal error:  Uncaught Error: Undefined class constant 'PRE_COMMAND_RUN' in /var/lib/monarc/fo/vendor/laminas/laminas-dependency-plugin/src/DependencyRewriterPlugin.php:63
Stack trace:
#0 /usr/share/php/Composer/EventDispatcher/EventDispatcher.php(378): Laminas\DependencyPlugin\DependencyRewriterPlugin::getSubscribedEvents()
#1 /usr/share/php/Composer/Plugin/PluginManager.php(239): Composer\EventDispatcher\EventDispatcher->addSubscriber(Object(Laminas\DependencyPlugin\DependencyRewriterPlugin))
#2 /usr/share/php/Composer/Plugin/PluginManager.php(205): Composer\Plugin\PluginManager->addPlugin(Object(Laminas\DependencyPlugin\DependencyRewriterPlugin))
#3 /usr/share/php/Composer/Installer/PluginInstaller.php(63): Composer\Plugin\PluginManager->registerPackage(Object(Composer\Package\CompletePackage), true)
#4 /usr/share/php/Composer/Installer/InstallationManager.php(173): Composer\Installer\PluginInstaller->install(Object(Composer\Repository\InstalledFilesystemRepository), Object(Composer\Package\CompletePackage))
#5 /usr/sha in /var/lib/monarc/fo/vendor/laminas/laminas-dependency-plugin/src/DependencyRewriterPlugin.php on line 63

Fatal error: Uncaught Error: Undefined class constant 'PRE_COMMAND_RUN' in /var/lib/monarc/fo/vendor/laminas/laminas-dependency-plugin/src/DependencyRewriterPlugin.php:63
Stack trace:
#0 /usr/share/php/Composer/EventDispatcher/EventDispatcher.php(378): Laminas\DependencyPlugin\DependencyRewriterPlugin::getSubscribedEvents()
#1 /usr/share/php/Composer/Plugin/PluginManager.php(239): Composer\EventDispatcher\EventDispatcher->addSubscriber(Object(Laminas\DependencyPlugin\DependencyRewriterPlugin))
#2 /usr/share/php/Composer/Plugin/PluginManager.php(205): Composer\Plugin\PluginManager->addPlugin(Object(Laminas\DependencyPlugin\DependencyRewriterPlugin))
#3 /usr/share/php/Composer/Installer/PluginInstaller.php(63): Composer\Plugin\PluginManager->registerPackage(Object(Composer\Package\CompletePackage), true)
#4 /usr/share/php/Composer/Installer/InstallationManager.php(173): Composer\Installer\PluginInstaller->install(Object(Composer\Repository\InstalledFilesystemRepository), Object(Composer\Package\CompletePackage))
#5 /usr/sha in /var/lib/monarc/fo/vendor/laminas/laminas-dependency-plugin/src/DependencyRewriterPlugin.php on line 63

Tried updating composer without success
Running on Ubuntu 18.04 lts freshly installed.

**Comment by cedricbonhomme** [comment]
Hi,

There is currently a bug with some version of composer which occurs when trying to install laminas/laminas-dependency-plugin. Especially with composer provided by Ubuntu and Debian packages. But not only.

We experienced the exact same issue yesterday when migrating our instances.

The fix we used is here:
https://github.com/monarc-project/MonarcAppFO/commit/4458f172c974382af1e2656f6116e44642d14863#diff-fcac54e391949c4fc49e6a8247f7d299R43-R46

It simply install a newer version of composer.
